internal/dag/gatewayapi_processor.go

```diff
@@ -1310,6 +1393,208 @@ func (p *GatewayAPIProcessor) computeHTTPRoute(route *gatewayapi_v1beta1.HTTPRou
+					continue
+				}
+
+				var err error
+				responseHeaderPolicy, err = headersPolicyGatewayAPI(filter.ResponseHeaderModifier, string(filter.Type))
+				if err != nil {
+					routeAccessor.AddCondition(gatewayapi_v1beta1.RouteConditionResolvedRefs, metav1.ConditionFalse, status.ReasonDegraded, fmt.Sprintf("%s on response headers", err))
+				}
+			case gatewayapi_v1alpha2.GRPCRouteFilterRequestMirror:
+				// Get the mirror filter if there is one. If there are more than one
+				// mirror filters, "NotImplemented" condition on the Route is set to
+				// status: True, with the "NotImplemented" reason.
+				if mirrorPolicy != nil {
+					routeAccessor.AddCondition(status.ConditionNotImplemented, metav1.ConditionTrue, status.ReasonNotImplemented, "GRPCRoute.Spec.Rules.Filters: Only one mirror filter is supported.")
+					continue
+				}
+
+				if filter.RequestMirror != nil {
+					mirrorService, cond := p.validateBackendObjectRef(filter.RequestMirror.BackendRef, "Spec.Rules.Filters.RequestMirror.BackendRef", KindGRPCRoute, route.Namespace)
+					if cond != nil {
+						routeAccessor.AddCondition(gatewayapi_v1beta1.RouteConditionType(cond.Type), cond.Status, gatewayapi_v1beta1.RouteConditionReason(cond.Reason), cond.Message)
+						continue
+					}
+					// If protocol is not set on the service, need to set a default one based on listener's protocol type.
+					setDefaultServiceProtocol(mirrorService, listener.listener.Protocol)
+					mirrorPolicy = &MirrorPolicy{
+						Cluster: &Cluster{
+							Upstream: mirrorService,
+						},
+					}
+				}
+			default:
+				routeAccessor.AddCondition(
+					gatewayapi_v1beta1.RouteConditionAccepted,
+					metav1.ConditionFalse,
+					gatewayapi_v1beta1.RouteReasonUnsupportedValue,
+					fmt.Sprintf("GRPCRoute.Spec.Rules.Filters: invalid type %q: only RequestHeaderModifier, ResponseHeaderModifier and RequestMirror are supported.", filter.Type),
+				)
+			}
+		}
+
+		// Priority is used to ensure if there are multiple matching route rules
+		// within an GRPCRoute, the one that comes first in the list has
+		// precedence. We treat lower values as higher priority so we use the
+		// index of the rule to ensure rules that come first have a higher
+		// priority. All dag.Routes generated from a single GRPCRoute rule have
+		// the same priority.
+		priority := uint8(ruleIndex)
+
+		// Get our list of routes based on whether it's a redirect or a cluster-backed route.
+		// Note that we can end up with multiple routes here since the match conditions are
+		// logically "OR"-ed, which we express as multiple routes, each with one of the
+		// conditions, all with the same action.
+		var routes []*Route
+
+		clusters, totalWeight, ok := p.GRPCClusters(route.Namespace, rule.BackendRefs, routeAccessor, listener.listener.Protocol)
+		if ok {
+			routes = p.clusterRoutes(matchconditions, requestHeaderPolicy, responseHeaderPolicy, mirrorPolicy, clusters, totalWeight, priority, nil)
+		}
```

Comment:
Nit, can simplify this a bit:

```suggestion
		clusters, totalWeight, ok := p.GRPCClusters(route.Namespace, rule.BackendRefs, routeAccessor, listener.listener.Protocol)
		if !ok {
			continue
		}
		
		routes = :p.clusterRoutes(matchconditions, requestHeaderPolicy, responseHeaderPolicy, mirrorPolicy, clusters, totalWeight, priority, nil)
```